tools/Customization/DevHome.FileExplorerSourceControlIntegration/Services/RepositoryTracking.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using DevHome.Common.Helpers;
+using DevHome.Common.Services;
+using Serilog;
+using Windows.Storage;
+
+namespace DevHome.FileExplorerSourceControlIntegration.Services;
+
+public class RepositoryTracking
+{
+    public RepoStoreOptions RepoStoreOptions
+    {
+        get; set;
+    }
+
+    public enum RepositoryChange
+    {
+        Added,
+        Removed,
+    }
+
+    private readonly FileService fileService;
+
+    private readonly object trackRepoLock = new();
+
+    private Dictionary<string, string> TrackedRepositories { get; set; } = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+    private readonly Serilog.ILogger log = Log.ForContext("SourceContext", nameof(RepositoryTracking));
+
+    public event Windows.Foundation.TypedEventHandler<string, RepositoryChange>? RepositoryChanged;
+
+    public DateTime LastRestore { get; set; }
+
+    public RepositoryTracking(string? path)
+    {
+        if (RuntimeHelper.IsMSIX)
+        {
+            RepoStoreOptions = new RepoStoreOptions
+            {
+                RepoStoreFolderPath = ApplicationData.Current.LocalFolder.Path,
+            };
+            log.Debug("Repo Store for File Explorer Integration created under ApplicationData");
+        }
+        else
+        {
+            RepoStoreOptions = new RepoStoreOptions
+            {
+                RepoStoreFolderPath = path ?? string.Empty,
+            };
+        }
+
+        fileService = new FileService();
+        RestoreTrackedRepositoriesFomJson();
+    }
+
+    public void RestoreTrackedRepositoriesFomJson()
+    {
+        lock (trackRepoLock)
+        {
+            var caseSensitiveDictionary = fileService.Read<Dictionary<string, string>>(RepoStoreOptions.RepoStoreFolderPath, RepoStoreOptions.RepoStoreFileName);
+
+            // No repositories are currently being tracked. The file will be created on first add to repository tracking.
+            if (caseSensitiveDictionary == null)
+            {
+                TrackedRepositories = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                log.Debug("Repo store cache has just been created");
+            }
+            else
+            {
+                TrackedRepositories = caseSensitiveDictionary.ToDictionary(entry => entry.Key, entry => entry.Value, StringComparer.OrdinalIgnoreCase);
+            }
+
+            LastRestore = DateTime.Now;

```

Comment:
NIT: use UTcNow.  An edge case can occur if using .Now.  For example, Daylight savings time occurred.  Or the user moved between times zones.